dace/subsets.py

```diff
@@ -45,24 +95,44 @@ def bounding_box_symbolic_positive(subset_a, subset_b, approximation=False) -> b
         return ValueError(f"A bounding box of dimensionality {len(min_elements_a)} cannot"
                           f" test covering a bounding box of dimensionality {len(min_elements_b)}.")
 
+    # NOTE: `nng()` is applied inside the loop.
+    simplify = lambda expr: symbolic.simplify_ext(expr)
+    no_simplify = lambda expr: expr
+
     for rb, re, orb, ore in zip(min_elements_a, max_elements_a, min_elements_b, max_elements_b):
-        # NOTE: We first test for equality, which always returns True or False. If the equality test returns
-        # False, then we test for less-equal and greater-equal, which may return an expression, leading to
-        # TypeError. This is a workaround for the case where two expressions are the same or equal and
-        # SymPy confirms this but fails to return True when testing less-equal and greater-equal.
+        # NOTE: Applying simplify takes a lot of time, thus we try to avoid it and try to do the test
+        #   first with the symbols we get and if we are unable to figuring out something, we run
+        #   simplify. Furthermore, we also try to postpone `nng()` as long as we can.
+        # NOTE: We have to use the `== True` test because of SymPy's behaviour. Otherwise we would
+        #   get an expression resulting in a `TypeError`.
+        # TODO: Figuring out why we use this two stage version, instead of `(y <= x) == True`.
```

Comment:
It used to be in the past - and intuitively I assume that still holds true - that an equality check has a much lower optimal runtime complexity than the less-than/greater-than comparison. Specifically for cases where equality is trivially given with no rearrangement / simplification. Note that I have not verified that this is in fact still the case.

TL;DR: speed